src/solver/conv_asm_implicit_gemm_wrw_gtc_dynamic_xdlops.cpp

```diff
@@ -340,144 +326,125 @@ FindImplicitGemmWrwGTCDynamicXdlopsKernel(const ConvolutionContext& ctx)
-                    continue;
-                for(j = 4; j > 1; j--)
-                {
-                    gemm_k_per_block = 1 << j;
-                    if(gemm_k % gemm_k_per_block != 0)
-                        continue;
-                    gemm_k_global_split = if_gemm_k_global_split(
-                        ctx, gemm_m_per_block, gemm_n_per_block, gemm_k_per_block);
-
-                    nxb               = 1;
-                    nxe               = 1;
-                    int tunable_index = -1;
-
-                    if((x * y * stride_h * stride_w == 1) && (ho * wo % 4 == 0))
-                    {
-                        nxb           = 4;
-                        nxe           = 0;
-                        tunable_index = find_tunable(tunables,
-                                                     gemm_m_per_block,
-                                                     gemm_n_per_block,
-                                                     gemm_k_per_block,
-                                                     gemm_k_global_split,
-                                                     nxb,
-                                                     nxe);
-                        if(tunable_index < 0 || tunable_index >= tunables.size())
-                        {
-                            nxb = 1;
-                            nxe = 1;
-
-                            // std::cout << gemm_m_per_block << ", " << gemm_n_per_block << ", " <<
-                            // gemm_k_per_block << std::endl;
-
-                            tunable_index = find_tunable(tunables,
-                                                         gemm_m_per_block,
-                                                         gemm_n_per_block,
-                                                         gemm_k_per_block,
-                                                         gemm_k_global_split,
-                                                         nxb,
-                                                         nxe);
-                        }
-                    }
-                    else
-                    {
-                        tunable_index = find_tunable(tunables,
-                                                     gemm_m_per_block,
-                                                     gemm_n_per_block,
-                                                     gemm_k_per_block,
-                                                     gemm_k_global_split,
-                                                     nxb,
-                                                     nxe);
-                    }
+                    const auto gemm_m_per_block = swap == 0 ? 1 << r : 1 << l;
+                    const auto gemm_n_per_block = swap == 0 ? 1 << l : 1 << r;
 
-                    if(tunable_index < 0 || tunable_index >= tunables.size())
+                    if(gemm_m % gemm_m_per_block != 0 || gemm_n % gemm_n_per_block != 0)
```

Comment:
The second condition here should be
 `if( c  %  (  gemm_n_per_block / (nxe == 0?1:nxe)  )  != 0)
    return false;`

Reply:
Do you mean add this condition? and `return false` ?

Reply:
As PR542[https://github.com/ROCmSoftwarePlatform/MIOpen/pull/542](url)